File: Stocks/SymbolsTableViewController.m
//
//  SymbolsTableViewController.m
//  Stocks
//
//  Created by Jonathan Wight on 6/9/14.
//  Copyright (c) 2014 schwa.io. All rights reserved.
//

#import "SymbolsTableViewController.h"

#import "SymbolViewController.h"
#import "StockModel.h"

@interface SymbolsTableViewController ()
@property (readwrite, nonatomic, copy) NSArray *symbols;
@end

@implementation SymbolsTableViewController

- (void)viewDidLoad
{
    [super viewDidLoad];

    self.symbols = [StockModel sharedInstance].symbols;
}


- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.symbols.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"CELL"];
    cell.textLabel.text = self.symbols[indexPath.row];
    return cell;
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    UITableViewCell *cell = sender;
    NSIndexPath *indexPath = [self.tableView indexPathForCell:cell];
    NSString *symbol = self.symbols[indexPath.row];

    SymbolViewController *theViewController = segue.destinationViewController;
    theViewController.symbol = symbol;

}

@end

File: Stocks/SymbolsTableViewController.swift
//
//  SymbolsTableViewController.swift
//  Stocks
//
//  Created by Jim Kubicek on 6/10/14.
//  Copyright (c) 2014 schwa.io. All rights reserved.
//

import UIKit

class SymbolsTableViewController: UITableViewController {

    let symbols = StockModel.sharedInstance().symbols as String[]

    // UIViewController

    init(nibName nibNameOrNil: String!, bundle nibBundleOrNil: NSBundle!) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }

    init(coder aDecoder: NSCoder!) {
        super.init(coder: aDecoder)
    }

    override func prepareForSegue(segue: UIStoryboardSegue!, sender: AnyObject!) {

        let cell = sender as UITableViewCell
        let indexPath = self.tableView.indexPathForCell(cell)
        let symbol = symbols[indexPath.row]

        let vc = segue.destinationViewController as SymbolViewController
        vc.symbol = symbol
    }

}

extension SymbolsTableViewController { // UITableViewDataSource

    override func tableView(tableView: UITableView!, numberOfRowsInSection section: Int) -> Int {

        return symbols.count

    }

    override func tableView(tableView: UITableView!, cellForRowAtIndexPath indexPath: NSIndexPath!) -> UITableViewCell! {

        let cell : UITableViewCell = tableView.dequeueReusableCellWithIdentifier("CELL", forIndexPath: indexPath) as UITableViewCell
        cell.textLabel.text = self.symbols[indexPath.row]
        return cell

    }

}




